### DISCUSSION TITLE
"Game mode"

### DISCUSSION THREAD

**Discussion by willwade** [dasher-project/redash#52] [UX]
I've just come back from a 2 hr training session for a new user on dasher. Now to do this I went straight to the Linux build. The Linux version is so much cleaner and error messages like the training text error doesn't come up - and switching between direct mode is far easier. But of course the real thing about the Linux version is it has the game mode. Some observations.

- Its neat. But the visual indicator is hard to get your head around. A pointy finger that maybe steers the user in the right direction - not just a line - maybe more useful?
- I'd like a "forced order" mode for this. So a way of never making a mistake. Like the UI would stop zooming - almost be paused - until you drove the letters through the right way. 

(If you are reading this & wondering "whats game mode?" see https://youtu.be/54AdL9KTjBs)

**Comment by jspatz1** [comment]
Thanks Will! 
I've been playing around with game mode myself and I know what some of my qualms are - but, it's great to hear from the users!

I suspect many of the March survey participants will not have experienced game mode recently or at all (true?) If so, perhaps it will be more effective to get user feedback on game mode with a quick questionnaire for those users who are trained on it in the coming weeks/months? 

What do you think? 

Also, the bulleted suggestions are wonderful. I'm curious, did this user ever miss more than one letter? I find that the indicator gets especially confusing when this happens.

**Reply by willwade** [reply]
True. I imagine only  a small handful will have even seen it.. 

Re: "quick questionnaire for those users who are trained on it in the coming weeks/months?" - yes - we could try that.. 

They did miss more than one letter. Needed quite a bit of support in general. That was though due to a) vision difficulties seeing the letters and b) mouse control (overshooting to the right making it go too fast). and maybe at times c) getting totally lost with it all!

**Comment by jspatz1** [comment]
Oh! And this is not urgent, but your above comment reminded me that I believe I am doing something wrong when trying to import training files into Dasher. I've copied a Wikipedia article into a Google Doc and downloaded as a .txt file. When I attempt to import it in Dasher, the zooming area disappears and I see the words "Training on user text 100%." Then I am unable to open a new file without closing and reopening Dasher. (is this the above mentioned training text error?)

Also, what did you mean above by "switching between direct mode"?

**Reply by willwade** [reply]
I have to admit - I'm a little lost on "training" text. I think training.txt is the same as your own corpus. So I think what you did is correct.. But I don't get why there is an error message - because when I look there is already a training text file..maybe I'm missing something too!

"Direct mode" is what dasher terms "on-screen keyboard". i.e. not type to a text box - but into whatever app has focus.

**Reply by willwade** [reply]
@Jspatz1 just remembered we have a similar game mode pencilled in for our other opensource aac project pasco https://github.com/AceCentre/pasco/issues/204 - the second idea might be relevant to dasher?
